fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -2250,18 +2227,47 @@ ACTOR Future<Void> TagPartitionedLogSystem::epochEnd(Reference<AsyncVar<Referenc
 
 			auto logSystem = makeReference<TagPartitionedLogSystem>(dbgid, locality, prevState.recoveryCount);
 
-			// If UNICAST is set, keep refreshing list TLog's recovery versions as maxEnd will vary
-			if (!SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
-				lastEnd = minEnd;
-				logSystem->recoverAt = minEnd;
-			} else {
-				logSystem->recoverAt = invalidVersion;
+			// If UNICAST is set, keep refreshing list TLog's recovery versions as maxEnd may change as new tLogs are
+			// locked.
+			logSystem->recoverAt = minEnd;
+			logSystem->maxRv = minEnd;
+
+			std::map<UID, Version> rvLogs;
+			if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
+
+				std::unordered_map<Version, int> versionRepCount;
+				for (auto& result : results) {
+					for (int i = 0; i < result.unknownCommittedVersions.size(); i++) {
+						Version k = std::get<0>(result.unknownCommittedVersions[i]);
+						versionRepCount[k]++;
```

Comment:
The assumption is that the default value of "int" is zero. Are you sure that that is guaranteed in all cases?

Reply:
For `std::map`, this is true.

Reply:
Ok, thanks for confirming!